consensus/README.md

```diff
@@ -0,0 +1,205 @@
+
+When the leader collects votes from more than two-thirds of the replicas, it progresses to the next consensus phase. This two-thirds rule is essential to achieve Byzantine Fault Tolerance (BFT), ensuring network resilience against faulty or malicious nodes.
+
+### Block Generation
+
+Block generation in the HotStuff consensus algorithm involves a series of interactive steps between the Leader and the Replica nodes. The steps are as follows:
+
+```mermaid
+sequenceDiagram
+    participant Leader
+    participant Replicas
+    Note over Leader,Replicas: Leader Election
+    Leader->>Replicas: Propose(block)
+    Note over Replicas: Validate proposed block
+    Replicas-->>Leader: Prepare(block)
+    Note over Leader: Receives Prepare messages from a quorum of Replicas
+    Leader->>Replicas: Pre-Commit(block, Prepare messages)
+    Note over Replicas: Validate Pre-Commit message
+    Replicas-->>Leader: Commit(block)
+    Note over Leader: Receives Commit messages from a quorum of Replicas
+    Leader->>Replicas: Notify(block, Commit messages)
+    Note over Replicas: Add block to local blockchain copy
+    Note over Leader,Replicas: New Leader Election
+```
+
+### Block Validation
+
+Every proposal made by the leader goes through a series of validation steps performed by the replicas.
+
+```mermaid
+graph TD
+    A[Receive Block proposal from leader]
+    B[Perform basic validation]
+    C[Check Block structure]
+    D[Validate quorum certificate]
+    E[Validate message signatures]
+    F[Validate optimistic threshold]
+    G[Block is Valid - apply block]
+    H[Block is Invalid - Reject the proposal]
+    J[Create and send vote message to the leader]
+    A-->B
+    B-->C
+    C-->D
+    D-->E
+    E-->F
+    F-->G
+    G-->J
+    B-.->H
+    C-.->H
+    D-.->H
+    E-.->H
+    F-.->H
+```
+
+### Consensus Lifecycle
+
+The consensus lifecycle begins with a new round that includes leader election, proposal creation, proposal validation, vote aggregation, and block commitment. The steps are as follows:
+
+```mermaid
+flowchart TD
+  A[Start New Round] --> |Elect Leader| L[Leader Election Module]
+  L --> D1[Leader]
+  L --> D2[Replica]
+  D1 --> E1[Create Proposals]
+  D2 --> E2[Validate Proposals]
+  E1 --> F1[Aggregate Votes]
+  E2 --> F2[Vote on Proposals]
+  F1 --> G1[Quorum and Commit Block]
+  F2 --> G2[Commit Block]
+  G1 --> J1[End Round]
+  G2 --> J1
+  J1 --> A
+```
+
+### State Sync
+
+State synchronization is crucial to ensure all participating nodes maintain a consistent and up-to-date view of the network state. It is especially important in a dynamic and decentralized network where nodes can join, leave, or experience intermittent connectivity. For an in-depth understanding of the state sync process and its current status, please refer to our [State Sync Protocol Design Specification](https://github.com/pokt-network/pocket/blob/main/consensus/doc/PROTOCOL_STATE_SYNC.md).
```

Comment:
NIT: Any chance you could follow the "best practices" I learnt about recently and see if we can reformat this (by moving this around) so it's easier to read: https://twitter.com/olshansky/status/1660499323317284864

If this is too hard, then no worries!

Reply:
@gokutheengineer we synched offline, but let's just merge in what we have and iterate on a diagram in a future PR!

Reply:
I did a quick fix that I think makes it better and easier to understand.